ngraph/core/src/op/dft.cpp

```diff
@@ -0,0 +1,259 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include <algorithm>
+#include <memory>
+#include <ngraph/validation_util.hpp>
+
+#include "itt.hpp"
+#include "ngraph/attribute_visitor.hpp"
+#include "ngraph/axis_set.hpp"
+#include "ngraph/axis_vector.hpp"
+#include "ngraph/op/constant.hpp"
+#include "ngraph/op/dft.hpp"
+#include "ngraph/op/util/op_types.hpp"
+#include "ngraph/runtime/host_tensor.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+NGRAPH_RTTI_DEFINITION(op::v7::DFT, "DFT", 7);
+
+op::v7::DFT::DFT(const Output<Node>& data, const Output<Node>& axes)
+    : Op({data, axes})
+{
+    constructor_validate_and_infer_types();
+}
+
+op::v7::DFT::DFT(const Output<Node>& data,
+                 const Output<Node>& axes,
+                 const Output<Node>& signal_size)
+    : Op({data, axes, signal_size})
+{
+    constructor_validate_and_infer_types();
+}
+
+bool op::v7::DFT::visit_attributes(AttributeVisitor& visitor)
+{
+    NGRAPH_OP_SCOPE(v7_DFT_visit_attributes);
+    return true;
+}
+
+std::shared_ptr<Node> op::v7::DFT::clone_with_new_inputs(const OutputVector& new_args) const
+{
+    NGRAPH_OP_SCOPE(v7_DFT_clone_with_new_inputs);
+    check_new_args_count(this, new_args);
+    NODE_VALIDATION_CHECK(
+        this, new_args.size() == 2 || new_args.size() == 3, "Number of inputs must be 2 or 3");
+
+    if (new_args.size() == 2)
+    {
+        return std::make_shared<op::v7::DFT>(new_args.at(0), new_args.at(1));
+    }
+
+    return std::make_shared<op::v7::DFT>(new_args.at(0), new_args.at(1), new_args.at(2));
+}
+
+void op::v7::DFT::validate()
+{
+    size_t num_of_inputs = get_input_size();
+
+    NODE_VALIDATION_CHECK(
+        this, num_of_inputs == 2 || num_of_inputs == 3, "DFT must have 2 or 3 inputs.");
+
+    element::Type input_et = get_input_element_type(0);
+    NODE_VALIDATION_CHECK(this,
+                          input_et == element::f32 || input_et == element::f16 ||
+                              input_et == element::bf16,
+                          "DFT input element type must be f32, f16, or bf16");
+
+    element::Type axes_et = get_input_element_type(1);
+    NODE_VALIDATION_CHECK(this,
+                          axes_et == element::i64 || axes_et == element::i32,
+                          "DFT axes element type must be i32 or i64");
+
+    PartialShape input_shape = PartialShape(get_input_partial_shape(0));
```

Comment:
Why do you create a copy of `PartialShape` everywhere? We don't modify the value of `input_shape` anywhere in this function, so we can use like this `const auto& input_shape = get_input_partial_shape(0);`

The same comment is applicable for other `PartialShapes` in this file

Reply:
Fixed.